optuna/integration/pytorch_lightning.py

```diff
@@ -39,6 +43,18 @@ def __init__(self, trial: optuna.trial.Trial, monitor: str) -> None:
 
         self._trial = trial
         self.monitor = monitor
+        self.is_ddp_backend = False
+
+    def on_init_start(self, trainer: Trainer) -> None:
+        self.is_ddp_backend = trainer.accelerator_connector.distributed_backend is not None
+        if self.is_ddp_backend:
+            if version.parse(pl.__version__) < version.parse("1.4.0"):
+                raise ValueError("PyTorch Lightning>=1.4.0 is required in DDP.")
+            if not isinstance(self._trial.study._storage, _CachedStorage):
+                raise ValueError(
+                    ":class:`~optuna.integration.PyTorchLightningPruningCallback`"
+                    " supports only :class:`~optuna.storages.RDBStorage` in DDP."
+                )
```

Comment:
This message is originally suggested in https://github.com/optuna/optuna/pull/2849#discussion_r702427394,
but I think error messages do not render the reStructured text. Can remove the markup?
```suggestion
                raise ValueError(
                    "optuna.integration.PyTorchLightningPruningCallback"
                    " supports only optuna.storages.RDBStorage in DDP."
                )
```